core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -0,0 +1,68 @@
+package cats
+package data
+
+import cats.std.list._
+//it needs a Functor
+//it needs semigroup - combine, filter
+
+/**
+  *  A data type which represents a single element (head) and some other
+  * structure (tail).
+  */
+final case class NonEmptyList[A](head: A, tail: List[A]) {
+
+  /**
+    * Return the head and tail into a single list
+    */
+  def unwrap: List[A] = head :: tail
+
+  /**
+    * remove elements not matching the predicate
+    */
+  def filter(p: A => Boolean): List[A] = {
+    val rest = tail.filter(p)
+    if(p(head)) head::rest else rest
+  }
+
+  /**
+    * Append another NonEmptyList
+    */
+  def combine(other: NonEmptyList[A]):NonEmptyList[A] = 
+    NonEmptyList(head, MonadCombine[List].combineK(tail, other.head::other.tail))
```

Comment:
Minor: this could use `other.unwrap` (or `other.toList` if we go with that name) instead of `other.head :: other.tail`, and there are a few other places in this file that could do the same.